app/cdap/components/PipelineDetails/ProfilesListView/ProfileCustomizePopover/ProfileCustomizeContent/index.js

```diff
@@ -80,13 +86,39 @@ export default class ProfileCustomizeContent extends PureComponent {
 
   componentWillUnmount() {
     this.customization = {};
-    this.latestValues = {};
   }
 
-  getDefaultValues() {
-    const { editablePropertiesFromProfile } = this.props;
+  getDefaultPropertiesFromProvisionerDetails = (provisionerConfigGroups) => {
+    const properties = provisionerConfigGroups.flatMap((configGroup) => configGroup.properties);
+    const defaultProperties = properties.map((property) => {
+      return {
+        name: property.name,
+        value: property['widget-attributes'].default ? property['widget-attributes'].default : '',
+        isEditable: true,
+      };
+    });
+    return defaultProperties;
+  };
+
+  combineAndFilterPropertiesWithDefaultValue = (originalProperties, defaultProperties) => {
+    let combinedProperties = [...originalProperties];
```

Comment:
We're using `PropertyRow` and `AbstractWidget` under the hood, so if you pass in `default` with the widget props, will that get picked up? Would save a lot of work here.

Reply:
Terrific suggestion! I have made the change

Reply:
hmm I took a look and I believe the `widget-attributes` are already sent to `ProfilePropertyRow`
<img width="454" alt="image" src="https://user-images.githubusercontent.com/98125204/234433801-1ea8050a-81bc-48fe-98da-9adce2152d66.png">

but it's not updating to default value when value is missing.

Reply:
And we also need to take care of customization value from runtimeargs, I believe it makes more sense to directly passing the value that we want to show.

Reply:
then it made me realize the fix can be much simpler...